pkg/registration/register.go

```diff
@@ -53,8 +68,26 @@ func Register(conn connection.Connection, regcode, hostname string, systemInform
 	}
 
 	credErr := creds.SetLogin(reg.Login, reg.Password)
+	if credErr != nil {
+		return reg.Id, credErr
+	}
+	regErr := mapRegIdToErr(reg.Id)
+	return reg.Id, regErr
```

Comment:
Note that currently this a breaking change which may or may not be important. It could be a good breaking change, in the sense that we do we may want to signal to end users that an error was actually encountered and the ID is not actually an ID

We can always omit returning the error here and consider some other options:
- Only surface constants as part of improving error handling
- Implement a public helper function like:
```go
func ErrorFromRegistrationId(regId int) error {
  // same strongly-typed errors in the current control flow
}
```